hw/top_earlgrey/data/ip/chip_rv_timer_testplan.hjson

```diff
@@ -16,5 +16,56 @@
       stage: V2
       tests: ["chip_sw_rv_timer_irq"]
     }
+    {
+      name: chip_sw_timer
+      desc: '''Verify the timeout interrupt assertion.
+
+            - Configure RV_TIMER to generate an interrupt after a set timeout.
+            - Issue a WFI to wait for the interrupt to trigger.
+            - Service the interrupt when it triggers; verify that it came from rv_timer.
+            - Verify that the interrupt triggered only after the timeout elapsed.
+            '''
+      stage: V2
+      si_stage: SV2
+      features: ["RV_TIMER.ENABLE", "RV_TIMER.CONFIG", "RV_TIMER.COMPARE", "RV_TIMER.INTERRUPT",
+                 "RV_TIMER.COUNTER"]
+      tests: ["chip_sw_rv_timer_irq"]
+    }
+    {
+      name: tick_configuration
+      desc: '''Verify that the timer can be configured to generate a system tick.
+
+            - Configure the timer to generate a tick of `T` microseconds long.
+            - Enable the timer.
+            - Start a busy loop of 1 second based on the `mcycleh` CSR.
+            - Read the number of ticks `n`.
+            - Verify that `n * T ~= 1 second` within 5% of tolerance.
+            - Reapeat for T in :
+                [1, 10, 100 and 1000]
+            '''
+      stage: V3
+      si_stage: SV2
+      features: ["RV_TIMER.ENABLE", "RV_TIMER.CONFIG", "RV_TIMER.COMPARE"]
+      tests: [""]
+    }
+    {
+      name: riscv_isa_csrs
+      desc: '''Verify that the timer counter and compare registers can be read and written via
+            the registers `mtime` and `mtimecmp` and that `mtime` will wrap to zero when the limit
+            is reached.
+
+            - Enable interrupts.
+            - Configure the compare register to zero by writing to `mtimecmp` CSR.
```

Comment:
I'm not a RISC-V expert, but don't we have to set `mtimecmp` to the counter value we expect the timer to get to at 500ms?

Reply:
You're right, but I'm trying to make this test also verify the wrap (maybe it's not relevant) by setting the `mtimecmp` to zero, and the `mtime` to a value that represents 500ms until it wraps to zero. Does that make sense?

Reply:
Oh, I see! Yes, this makes perfect sense (and I can't see any way for it to be phrased more clearly - the problem was at my end :-D)